### PR TITLE
Add Lighthouse card & tab

### LINKED ISSUE #2605: Lighthouse plugin widgets/cards & entity integration
I'd like to add some features to the Lighthouse plugin and was hoping to get some input before I get too far into implementation. 

## Feature Suggestion

1. Add a Last Audit widget/card (not too sure what the preferred naming is for these) which could be displayed on the overview page for website components.
2. Add a Lighthouse tab to website components where detailed audit information is displayed

## Possible Implementation

Websites are identified in the lighthouse audit service by the url so I was thinking we could )optionally annotate website components with their url so that we can get audit information within the context of the catalog. 
E.g. `lighthouse.com/website-url: <website_url>`

For the widget, I'm thinking of perhaps two (so users can choose), the first of which something like this:
image

And the second perhaps using `<GaugeProgress />` elements to display scores in each category? For reference, something like [this](https://backstage.io/storybook/?path=/story/progress-card--default) but multiple guages in one card I'm thinking.

The tab would simply display the same content as the lighthouse plugin page when you click through to a particular website. Ideally we could just render the component which renders the content below in the tab:
image

### PR DESCRIPTION
**Pull request by iain-b**
fixes #2605 
- Add a lighthouse card to be displayed on the website overview
- Add a lighthouse tab and embed lighthouse content

Screencast: https://www.loom.com/share/6c7e056c83a94043bd5091bd1fd34cdf

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by stefanalund** [comment]
Personally I would prefer a UI with a list of "rings", similar to what you get inside the plugin. This is not a strong preference, but there are many lists in Backstage already and a "ring" design would immediately be recognised as Lighthouse.

**Comment by iain-b** [comment]
@stefanalund I had originally thought to do something like that but the only "ring" components I found in the storybook were quite large:
image

So I thought perhaps to start with something more compact. I could add another, larger widget using these rings (GaugeProgress)? and or create an issue for smaller gauge progress components to "shrink" that component.

Just to clarify also, those rings in the lighthouse audit are rendered in an iframe and aren't something we have access to directly; they're not in the backstage codebase.

**Comment by stefanalund** [comment]
@iain-b yeah, I know they are part of an iframe and that there are no good existing Storybook components. There might be a case for making smaller variants of those Gauges though.

I'm totally fine with going with the existing design and leaving my suggestions as future work. Please create an issue if you go down that route. 🥇

**Comment by stefanalund** [comment]
@iain-b excellent work. There is a conflict now though.

File: plugins/catalog/src/index.ts [hunk 1]
```diff
@@ -20,6 +20,6 @@ export * from './api/types';
 export * from './routes';
 export { useEntityCompoundName } from './components/useEntityCompoundName';
 export { Router } from './components/Router';
-export { useEntity } from './hooks/useEntity';
+export { useEntity, EntityContext } from './hooks/useEntity';
```
Comment (iain-b) [inline plugins/catalog/src/index.ts:23]: I should probably highlight this as it's likely unexpected. I found I needed to export this to test a plugin component which uses useEntity().

File: plugins/lighthouse/src/Router.tsx [hunk 2]
```diff
@@ -15,11 +15,19 @@
...
+import AuditView, { AuditViewContent } from './components/AuditView';
+import CreateAudit, { CreateAuditContent } from './components/CreateAudit';
+import { Entity } from '@backstage/catalog-model';
+import { LIGHTHOUSE_WEBSITE_URL_ANNOTATION } from '../constants';
+import { AuditListForEntity } from './components/AuditList/AuditListForEntity';
+import { EmptyState } from '@backstage/core';
+
+export const isPluginApplicableToEntity = (entity: Entity) =>
+  Boolean(entity.metadata.annotations?.[LIGHTHOUSE_WEBSITE_URL_ANNOTATION]) ||
+  entity.metadata.name === 'marketing-site';
```
Comment (nikek) [inline plugins/lighthouse/src/Router.tsx:30]: This looks very specific, is there a valid reason to keep it?

**Review by nikek — COMMENTED** [review]
What is the `plugins/backstage-plugin-travis-ci` submodule for in this PR?

File: plugins/lighthouse/src/Router.tsx (hunk 2 above)
Comment (iain-b) [reply plugins/lighthouse/src/Router.tsx:30]: Nope 😬

**Comment by iain-b** [comment]
> What is the `plugins/backstage-plugin-travis-ci` submodule for in this PR?

I was testing some unpublished changes to that plugin. I deleted the contents of that folder but not the folder itself 🤦

**Review by nikek — APPROVED** [review]

**Pull request merged**